Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/loader/impl/post_comment/CommentSpanUpdater.kt

```diff
@@ -0,0 +1,161 @@
+package com.github.adamantcheese.chan.core.loader.impl.post_comment
+
+import android.graphics.Bitmap
+import android.text.SpannableStringBuilder
+import android.text.Spanned
+import android.text.style.ImageSpan
+import com.github.adamantcheese.chan.core.model.Post
+import com.github.adamantcheese.chan.core.settings.ChanSettings
+import com.github.adamantcheese.chan.utils.AndroidUtils
+import com.github.adamantcheese.chan.utils.AndroidUtils.sp
+import com.github.adamantcheese.chan.utils.BackgroundUtils
+import com.github.adamantcheese.chan.utils.putIfNotContains
+import java.util.*
+import kotlin.math.abs
+
+internal object CommentSpanUpdater {
+    private const val TAG = "CommentSpanUpdater"
+    private val comparator = GroupedSpanUpdatesMapComparator()
+
+    fun updateSpansForPostComment(
+            post: Post,
+            spanUpdateBatchList: List<SpanUpdateBatch>
+    ): Boolean {
+        BackgroundUtils.ensureBackgroundThread()
+        val ssb = synchronized(this) { SpannableStringBuilder(post.comment) }
+        val groupedSpanUpdates = groupSpanUpdatesByOldSpans(spanUpdateBatchList)
+
+        // Since we are inserting new text, old spans will become incorrect right after the first
+        // insertion. To fix that we need to have a counter of extra characters added into the ssb
+        var offset = 0
+        var updated = false
+
+        groupedSpanUpdates.entries.forEach { (postLinkableSpan, invertedSpanUpdateBatchList) ->
+            invertedSpanUpdateBatchList.forEach { invertedSpanUpdateBatch ->
+                val start = postLinkableSpan.start + offset
+                val end = postLinkableSpan.end + offset
+                val originalLinkUrl = ssb.substring(start, end)
+
+                val formattedLinkUrl = formatLinkUrl(
+                        originalLinkUrl,
+                        invertedSpanUpdateBatch.extraLinkInfo
+                )
+
+                if (originalLinkUrl == formattedLinkUrl) {
+                    return@forEach
```

Comment:
Probably should only add the icon here.

Reply:
Actually this check is pointless since I always add two spaces "  " to every link to prepend the icon.